go/cmd/dolt/commands/tblcmds/import.go

```diff
@@ -724,26 +725,47 @@ func NameAndTypeTransform(row sql.Row, rowOperationSchema sql.PrimaryKeySchema,
 	row = applyMapperToRow(row, rowOperationSchema, rdSchema, nameMapper)
 
 	for i, col := range rowOperationSchema.Schema {
-		switch col.Type {
-		case sql.Boolean, sql.Int8, sql.MustCreateBitType(1): // TODO: noms bool wraps MustCreateBitType
-			switch row[i].(type) {
-			case int8:
-				val, ok := stringToBoolean(strconv.Itoa(int(row[i].(int8))))
-				if ok {
-					row[i] = val
-				}
-			case string:
-				val, ok := stringToBoolean(row[i].(string))
-				if ok {
-					row[i] = val
-				}
-			case bool:
-				row[i] = row[i].(bool)
-			}
+		// Check if this a string that can be converted to a boolean
+		val, ok := detectAndConvertToBoolean(row[i], col.Type)
+		if ok {
+			row[i] = val
+			continue
 		}
 
 		switch col.Type.(type) {
 		case sql.StringType:
+		// Bit types need additional verification due to the differing values they can take on. "4", "0x04", b'100' should
+		// be interpreted in the correct manner.
+		case sql.BitType:
+			colAsString, ok := row[i].(string)
+			if !ok {
+				return nil, fmt.Errorf("error: column value should be of type string")
+			}
+
+			// Check if the column can be parsed an uint64
+			val, err := strconv.ParseUint(colAsString, 10, 64)
+			if err == nil {
+				row[i] = val
+				continue
+			}
+
+			// Check if the column is of type b'110'
+			re := regexp.MustCompile(`(?m)b\'(\d+)\'`)
+			groups := re.FindStringSubmatch(colAsString)
+			if len(groups) > 1 {
+				val, err = strconv.ParseUint(groups[1], 2, 64)
+				if err == nil {
+					row[i] = val
+					continue
+				}
+			}
+
+			// Check if the column can be parsed as a hex string
+			numberStr := strings.Replace(colAsString, "0x", "", -1)
+			val, err = strconv.ParseUint(numberStr, 16, 64)
+			if err == nil {
+				row[i] = val
+			}
```

Comment:
if none of these worked and `row[i]` still isn't set, we should probably return an error instead of continuing execution and making the problem harder to find.

Reply:
Hmmm I see your point. Eventually the error does get caught at the insert node. I could see that being a potential UX problem.